### ISSUE TITLE
Connect with compute instance

### ISSUE THREAD

**Issue by bostdiek** [microsoft/vscode-tools-for-ai#1509]
Issue Type: <b>Bug</b>

I see my compute instance in the Azure Machine Learning extension. However, when I try to connect on my Mac in VSCode with Command + shift + P then connect to compute instance, I cannot connect.

The dropdown menu shows my subscription, which I click on, but then it does not see any of my compute instances and only gives me the option to create new ones.

Extension version: 0.6.28
VS Code version: Code 1.65.2 (Universal) (c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1, 2022-03-10T14:33:49.188Z)
OS version: Darwin x64 21.4.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5287U CPU @ 2.90GHz (4 x 2900)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 4, 5|
|Memory (System)|16.00GB (0.59GB free)|
|Process Argv|--crash-reporter-id f1e6bc6b-a3ee-44eb-8375-494fa153fbe5|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vsclayoutctrt:30451275
dsvsc009:30452663
pythonvsnew555:30457759
vscscmwlcmt:30438805
vscgsvid1:30447480
cppdebug:30451566
pynewfile477cf:30450039

```

</details>

**Comment by shsuman** [comment]
@bostdiek, Can you please provide me your e-mail id, so that I can send you our pre-release versions, and you try that out. My believe is that should mitigate your issue.

**Comment by bostdiek** [comment]
You could send it to my work email at ***@***.*** Thanks!

**Comment by shsuman** [comment]
@bostdiek  Your email id got redacted somehow. Can you email me on my id, its <my_github_username>@microsoft.com

**Comment by shsuman** [comment]
@bostdiek Please let me know if you are able to connect to the CI using he build I sent you

**Comment by bostdiek** [comment]
I didn't see anything sent to me, but I am able to connect now. So I guess the issue is resolved.

**Comment by bostdiek** [comment]
I am having issues with this again:
This image shows that my Azure Account extension does see the compute instance, however, I cannot connect to it (there is no option).
![image](https://user-images.githubusercontent.com/10420927/161622267-682bb6e4-27d0-488d-a3fb-e51393a49cbb.png)

Similarly, if I navigate to the option with Command + shift + P, "connect to compute instance", I can click on the subscription and Azure ML, but then it shows no compute instances.
![image](https://user-images.githubusercontent.com/10420927/161622825-1b01dab3-43a5-40a6-9083-594b9117c557.png)

**Comment by shsuman** [comment]
@bostdiek We are aware of the issue and if everything goes well, we will release a fix tomorrow. In the meantime, if you would like to get unblocked you can try one of the following:

* If you have a multi tenant account and your subscription is assigned to a tenant other than the default, follow the instructions [here](https://github.com/microsoft/vscode-azure-account/wiki/Troubleshooting#configure-your-tenant-in-the-azure-account-extension)
* Alternatively,  you can downgrade the extension to the previous version by following the instructions [here](https://code.visualstudio.com/updates/v1_30#_install-previous-versions)

**Comment by bostdiek** [comment]
Thank you! It was the tenant issue that is affecting me. The workaround is
very nice to know about. Cheers!

**Comment by shsuman** [comment]
@bostdiek Could you please upgrade the Azure ML extension to 0.8.1 and confirm whether that resolves your issue ?

**Comment by bostdiek** [comment]
@shsuman, I updated to Azure ML extension 0.8.1, then logged out of my accounts and retried logging in. It seems to have problems finding the subscription for my non-default accounts. As I saw a new error code, I registered a new issue (https://github.com/microsoft/vscode-tools-for-ai/issues/1532)

**Comment by shsuman** [comment]
@bostdiek Just to confirm, you cannot see your subscription in our tree view after logging in?

You can open the tree view by clicking the Azure icon in the Activity bar (Shift + Alt +A).

![image](https://user-images.githubusercontent.com/44413557/162000376-a79f8680-e306-42a6-a50b-12a4d98d71c4.png)

**Comment by bostdiek** [comment]
Correct, if I change tenant, I can see the one subscription, but not the
one for my standard account.
[image: image.png]
The first image is the tenant for one of our customers. The second image is
my personal account.
[image: image.png]
Both of these are registered to my work email address, but I have to use
Command + Shift + P > Azure: Select Tenant to change between these. In the
older extensions, I saw both here at the same time.

**Comment by shsuman** [comment]
@bostdiek I am going to close this issue and keep your other issue open. I have asked the Az Account team about this behavior. Will update you in the other issue
